Treat MMCA version 4 cards a version 3 cards.  It seem to work for me,
but it may break things horribly.

So you may not want to use this patch.

Index: linux-2.6.14/drivers/mmc/mmc.c
===================================================================
--- linux-2.6.14.orig/drivers/mmc/mmc.c
+++ linux-2.6.14/drivers/mmc/mmc.c
@@ -495,6 +495,7 @@ static void mmc_decode_cid(struct mmc_ca
 
 		case 2: /* MMC v2.0 - v2.2 */
 		case 3: /* MMC v3.1 - v3.3 */
+		case 4: /* MMC Plus? */
 			card->cid.manfid	= UNSTUFF_BITS(resp, 120, 8);
 			card->cid.oemid		= UNSTUFF_BITS(resp, 104, 16);
 			card->cid.prod_name[0]	= UNSTUFF_BITS(resp, 96, 8);
